src/control/server/config/server.go

```diff
@@ -502,37 +508,112 @@ func (cfg *Server) SetNrHugepages(log logging.Logger, mi *common.MemInfo) error
 		log.Debug(msg)
 	}
 
-	if cfgTargetCount <= 0 {
-		return nil // no nvme, no hugepages required
-	}
+	return
+}
 
-	if cfg.DisableHugepages {
-		return FaultConfigHugepagesDisabledWithBdevs
+func (cfg *Server) getMinMaxNrHugepages(log logging.Logger, hpSizeKiB int) (int, int, error) {
+	cfgTargetCount, sysXSCount := cfg.getTgtCounts(log)
+
+	if cfgTargetCount == 0 {
+		return 0, 0, nil
 	}
 
 	// Calculate minimum number of hugepages for all configured engines.
-	minHugepages, err := storage.CalcMinHugepages(mi.HugepageSizeKiB, cfgTargetCount+sysXSCount)
+	minHugepages, err := storage.CalcMinHugepages(hpSizeKiB, cfgTargetCount+sysXSCount)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	maxTgtCount := largeTargetCount
+	if sysXSCount > 0 {
+		maxTgtCount += largeSysXSCount
+	}
+	maxHugepages, err := storage.CalcMinHugepages(hpSizeKiB, maxTgtCount)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	var msgSysXS string
+	if sysXSCount > 0 {
+		msgSysXS = fmt.Sprintf(" and %d/%d sys-xstreams", sysXSCount, largeSysXSCount)
+	}
+	log.Tracef("calculated min/max %d/%d nr_hugepages based on %d/%d targets%s",
+		minHugepages, maxHugepages, cfgTargetCount, largeTargetCount, msgSysXS)
+
+	if minHugepages > maxHugepages {
+		log.Debugf("config hugepage requirements exceed normal maximum")
```

Comment:
Should this be logged at NOTICE level? Who is it for?

Reply:
this is just a debug message because the user doesn't need to do anything about it, it is an indication that the configuration requires more huge pages than the normal maximum

Reply:
The numbers are included in the trace level message, it doesn't indicate a problem only that the calculated value based on the config exceeds the normal upper bound used for the initial allocation.

Reply:
In theory someone could see this debug-level message without seeing the trace-level message though. I think you could either make both messages the same log level, or include the min and max values in this message.